src/Services/DatabaseAuditLog.cs

```diff
@@ -0,0 +1,46 @@
+using System.Security.Claims;
+using IdentityServerConfig.Data;
+using IdentityServerConfig.Models;
+using Microsoft.AspNetCore.Components.Authorization;
+
+namespace IdentityServerConfig.Services;
+
+public class DatabaseAuditLog : IAuditLog
+{
+    private readonly AuditContext _auditContext;
+    private readonly AuthenticationStateProvider _authenticationStateProvider;
+    private readonly ILogger _logger;
+
+    public DatabaseAuditLog(AuditContext auditContext, AuthenticationStateProvider authenticationStateProvider, ILogger<DatabaseAuditLog> logger)
+    {
+        _auditContext = auditContext;
+        _authenticationStateProvider = authenticationStateProvider;
+        _logger = logger;
+    }
+    
+    public void Log(string action, IDictionary<string,string> data)
+    {
+        var user = _authenticationStateProvider.GetAuthenticationStateAsync().Result.User;
+        var userId = user.Claims.FirstOrDefault(c => c.Type.Equals(ClaimTypes.NameIdentifier))?.Value;
+        var userName = user.Claims.FirstOrDefault(c => c.Type.Equals("name"))?.Value;
+
+        if (userId == null || userName == null)
+            throw new Exception("User not found");
```

Comment:
Perhaps an `InvalidOperationException` would be more appropriate here. The top-level `Exception` type is typically not used in code.

Reply:
changed it in 5a020fb